tests-blackbox/query/filter/index.ts

```diff
@@ -265,12 +277,22 @@ function processValidation(
 				return found;
 			}
 		} else {
+			const valueToCheck = stripArray ? stripArrayFromValue(data, currentKey) : get(data, currentKey);
 			if (assert) {
-				expect(processValidation(get(data, currentKey), keys.join('.'), filter, possibleValues, assert)).toBe(true);
+				expect(processValidation(valueToCheck, keys.join('.'), filter, possibleValues, assert, stripArray)).toBe(true);
 				return true;
 			} else {
-				return processValidation(get(data, currentKey), keys.join('.'), filter, possibleValues, assert);
+				return processValidation(valueToCheck, keys.join('.'), filter, possibleValues, assert, stripArray);
 			}
 		}
 	}
 }
+
+// Added for inconsistencies in JSON queries where some vendors return the fetched result in array format
```

Comment:
Would that be something we can intercept and process in the actual implementation? Or wouldn't that make sense? 🤔